packages/components/src/components/ResizableBox/ResizableBox.tsx

```diff
@@ -0,0 +1,209 @@
+import styled from 'styled-components';
+import { useEffect, useRef, useState } from 'react';
+
+export type ResizableBoxProps = {
+    children: React.ReactNode;
+    directions: Array<"top" | "left" | "right" | "bottom">; 
+    isLocked?: boolean; // if true, it's not possible to resize the component
+    width?: number;
+    minWidth?: number;
+    maxWidth?: number;
+    height?: number;
+    minHeight?: number;
+    maxHeight?: number;
+};
+
+const reactiveAreaWidth = '15px';
+
+type ResizersProps = {
+    $width?: number;
+    $minWidth?: number;
+    $maxWidth?: number;
+    $height?: number;
+    $minHeight?: number;
+    $maxHeight?: number;
+    x?: number; // TODO not really used
+    y?: number;
+};
+
+const Resizers = styled.div<ResizersProps>(({ $width, $minWidth, $maxWidth, $height, $minHeight, $maxHeight, x, y }) => `
+        ${$width ? `width: ${$width}px;` : 'width: auto;'};
+        ${$minWidth && `min-width: ${$minWidth}px;`};
+        ${$maxWidth && `max-width: ${$maxWidth}px;`};
+        ${$height ? `height: ${$height}px;` : 'height: auto;'};
+        ${$minHeight && `min-height: ${$minHeight}px;`};
+        ${$maxHeight && `max-height: ${$maxHeight}px;`};
+        // ${x && `left: ${x}px;`};
+        // ${y && `top: ${y}px;`};
+        box-sizing: border-box;
+        position: relative;
+
+        #top-div, #left-div, #right-div, #bottom-div {
```

Comment:
That's good to know! So in this case, I must reconsider this part of the implementation.

Reply:
@jvaclavik Now I can see I should have implemented the box a bit different way: not to add ids to those divs to style them like that (+ adding event listeners as I did) but kinda different way + use React `onMouseOver` with button element for those invisible divs and/or `onDrag` event handler on the div with the content being resized. I think this could solve more problems I am facing now. WDYT? Hopefully I can continue working on this amazing first task! :)